cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -133,6 +133,9 @@ enum class splitter_mode {
     random
 };
 
+/// Available engine methods for building trees
+enum class df_engine_method { mt2203, mcg59, philox4x32x10, mt19937, mrg32k3a };
```

Comment:
I would be beneficial to have this enum in some common place like dal/engine_method.hpp or dal/algo/engine_method.hpp to reduce potential code duplication.

Because the same methods can be used by other algorithms which use RNGs, like random initialization of K-means.

I'd also move `convert_engine_method` into backend/primitives/rng folder.